examples/s3-import-mysql/README.md

```diff
@@ -0,0 +1,103 @@
+# MySQL S3 Import Example
+
+Configuration in this directory creates set of RDS resources including DB instance, DB subnet group and DB parameter group where the database itself is imported from a MySQL Percona Xtrabackup stored in S3.
+
+## Usage
+
+To run this example you need to execute:
+
+```bash
+$ terraform init
+$ terraform plan
+$ terraform apply
+```
+
+## Testing
+
+In order to test this example, you will need a database backup in order to upload to S3 and import into the module. The steps outlined below should suffice for creating a backup that can be uploaded to S3 and imported for the sake of testing and verifying module functionality/changes.
+
+1. Create database container
+
+```bash
+$ docker run -d --name percona-server-mysql-8.0.20 -e MYSQL_ROOT_PASSWORD=root percona/percona-server:8.0.20
+$ docker exec -it percona-server-mysql-8.0.20 bash
+$ mysql -u root -p
+```
+
+2. Once logged into container and database, create database and user used by RDS
+
+```sql
+CREATE DATABASE s3Import;
+CREATE USER 's3_import_user'@'localhost' IDENTIFIED BY 'YourPwdShouldBeLongAndSecure!';
+GRANT ALL PRIVILEGES ON * . * TO 's3_import_user'@'localhost';
+FLUSH PRIVILEGES;
+```
+
+3. Use Percona Xtrabackup container to dump database and upload to S3
+
+```bash
+$ docker run --name percona-xtrabackup-8.0 --mount type=bind,src=/tmp/backup,dst=/backup --volumes-from percona-server-mysql-8.0.20 percona/percona-xtrabackup:8.0 xtrabackup --backup --data-dir=/var/lib/mysql --target-dir=/backup --user=root --password=root
+$ s3 sync /tmp/backup/ s3://s3-import-<UPDATE-NAME>/
```

Comment:
Can you please make a tiny SQL backup file in this example dir, create S3 bucket and upload it to the new bucket using [aws_s3_bucket_object](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket_object) resource, and then reference it in a test?

The goal is to run just "terraform apply" to get end-to-end examples to work.

Reply:
yep - I can add that

Reply:
FYI - there are quite a few files (you can see them in https://github.com/terraform-aws-modules/terraform-aws-rds/pull/289/commits/6be766667530a18d402fbc99efbc486b122bfe79) so I couldn't use the s3 bucket object resource. I did try with the `fileset()` but that doesn't support recursively adding everything so I had to go with a local-exec to make it work

Reply:
I see, I didn't know there would be so many files needed for the small demo. Thought there would be just 1-3 :)

Anyway, your solution with `unzip and s3 sync` looks perfect as the working example.